packages/react/src/framework-delegate.tsx

```diff
@@ -1,33 +1,38 @@
 import { FrameworkDelegate } from '@ionic/core/components';
 import { createPortal } from 'react-dom';
 
+type ReactComponent = (props?: any) => JSX.Element;
+
 export const ReactDelegate = (
-  addView: (view: React.ReactPortal) => void,
-  removeView: (view: React.ReactPortal) => void
+  addView: (view: React.ReactElement) => void,
+  removeView: (view: React.ReactElement) => void
 ): FrameworkDelegate => {
-  let Component: React.ReactPortal;
+  const refMap = new WeakMap<ReactComponent, React.ReactElement>();
 
   const attachViewToDom = async (
     parentElement: HTMLElement,
-    component: () => JSX.Element,
+    component: ReactComponent,
     propsOrDataObj?: any,
     cssClasses?: string[]
   ): Promise<any> => {
     const div = document.createElement('div');
     cssClasses && div.classList.add(...cssClasses);
     parentElement.appendChild(div);
 
-    Component = createPortal(component(), div);
+    const componentWithProps = component(propsOrDataObj);
+    const hostComponent = createPortal(componentWithProps, div);
 
-    Component.props = propsOrDataObj;
+    refMap.set(component, hostComponent);
```

Comment:
Why do we need the map?

Reply:
One instance of the delegate is constructed for one instance of `IonNav`, but `IonNav` can attach multiple views to the DOM. 

The map allows us to keep track of all instances of the individual views that were pushed inside the Nav. We can then do a look-up against the map to find the correct instance to remove the view from the DOM. 

In the previous implementation, the value of `Component` would be reset when a new view was pushed, which prevented the previous view from ever being removed.

Alternatively you could use an object instead of a map here, but since we are associating to a React component, a `WeakMap` would be my recommendation here so that it does not interfere with garbage collection.